### DISCUSSION TITLE
Github desktop user here, should I learn Git terminal commands?

### DISCUSSION THREAD

**Discussion by max-torch** [community/community#22824] [New to GitHub]
<p>Been using Github desktop for a while. So far I can do everything I want - merge, commit, rebase, push, pull, create and publish branches, switch between branches, switch repositories, clone, make diff files, see history and changes, compare branches. If there is just one particular file I modified but I want to undo the change I download the file from the online browser repository and paste it back into the local repository.</p>
<p>I feel comfortable doing all those functions with the desktop app, so should I learn Git terminal commands and what for?<br>
Asides from just entering commands a line at a time at a terminal can I make a script “macro” that will execute series of git commands?<br>
And if yes what kind of “macros” could come in handy?</p>
<p>P.S. First post.</p>

**Comment by ernest-phillips** [comment, marked as answer]
<p>Hi <a class="mention" href="/max-torch">@max-torch</a> !</p>
<p>I think it’s great that you found a workflow that suits your needs! Learning git for the command line can be helpful, but may not necessary depending on the problem you are trying to solve.</p>
<p>There are things that can be done via the command-line that simply aren’t possible via Desktop that  <em>can</em>  come in handy if you get into a bit of a bind. However, GitHub Desktop is great for the normal day-to-day application and even some intermediate tasks as you have noted.</p>
<p>This post may help to clarify the differences between the Command Line Interface (CLI) and the Desktop application.<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/5134583/181073256-c031c3b5-c29a-4c80-8415-631f90e3573d.png" class="site-icon" width="64" height="64">
      <a href="https://www.slant.co/versus/1501/13488/~the-command-line_vs_github-desktop" target="_blank" rel="noopener">Slant</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/341;"><img src="https://user-images.githubusercontent.com/5134583/181073258-e5030630-933b-4e80-8c65-ead204e9113d.png" class="thumbnail" width="690" height="341" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7129aff70d25e2101d47dd8f8f6524b78ecc119a_2_690x341.png, https://aws1.discourse-cdn.com/github/original/2X/7/7129aff70d25e2101d47dd8f8f6524b78ecc119a.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/7/7129aff70d25e2101d47dd8f8f6524b78ecc119a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7129aff70d25e2101d47dd8f8f6524b78ecc119a_2_10x10.png"></div>

<h3><a href="https://www.slant.co/versus/1501/13488/~the-command-line_vs_github-desktop" target="_blank" rel="noopener">Slant - The Command Line vs GitHub Desktop detailed comparison as of 2020</a></h3>

<p>When comparing The Command Line vs GitHub Desktop, the Slant community recommends The Command Line for most people. In the question "What are the best Git clients for macOS?" The Command Line is ranked 1st while GitHub Desktop is ranked 11th</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by max-torch** [comment]
<p>Thank you so much for answering my inquiry. For some reason I couldn’t quite find an article that explains it like what you’ve shared.<br>
I started trying Git using the command line. I’m not yet quite aware of something that I would want to do for my projects that I haven’t done in the desktop client, but that’s probably because I have gotten so used to the desktop client for my workflow. I’ve decided I’ll keep using the CLI for a while until it really clicks.<br>
Opinion so far:<br>
The desktop client has been glorious for getting just some random person to contribute to a Git project like I can even see how I can easily explain to my parents how to contribute to a Git repo using the desktop so it is definitely not something to take for granted for existing.</p>

**Comment by ernest-phillips** [comment]
<p>Hi <a class="mention" href="/max-torch">@max-torch</a>,<br>
I am so glad to hear about your success with using the desktop application. Many folks new to GitHub will start with the desktop tool. I am one of those people :raised_hand:<br>
It’s super cool that you have been able to share what you’ve learned with your parental units. I hope they learn to enjoy the tool as much as you have.<br>
If you decide to dive deeper into the CLI, and need some help or encouragement, please come back to the Community. We will be happy to help!</p>
